spectral_cube/dask_spectral_cube.py

```diff
@@ -707,6 +707,54 @@ def argmin(self, axis=None, **kwargs):
         """
         return self._compute(da.nanargmin(self._get_filled_data(fill=np.inf), axis=axis))
 
+    @ignore_warnings
+    def statistics(self):
+        """
+        Return a dictinary of global basic statistics for the data.
+
+        This method is designed to minimize the number of times each chunk is
+        accessed. The statistics are computed for each chunk in turn before
+        being aggregated.
+
+        The names for each statistic are adopted from CASA's ia.statistics
+        (see https://casa.nrao.edu/Release4.1.0/doc/CasaRef/image.statistics.html)
+        """
+
+        data = self._get_filled_data(fill=np.nan)
+
+        try:
+            from bottleneck import nanmin, nanmax, nansum
+        except ImportError:
+            from numpy import nanmin, nanmax, nansum
+
+        def compute_stats(chunk, *args):
+            return np.array([np.sum(~np.isnan(chunk)),
+                             nanmin(chunk),
+                             nanmax(chunk),
+                             nansum(chunk),
+                             nansum(chunk * chunk)])
+
+        with dask.config.set(**self._scheduler_kwargs):
+            results = da.map_blocks(compute_stats, data.reshape(-1)).compute()
+
+        count_values, min_values, max_values, sum_values, ssum_values = results.reshape((-1, 5)).T
+
+        stats = {'npts': count_values.sum(),
+                 'min': min_values.min() * self._unit,
```

Comment:
This should use `self.unit`, I think. When `self.unit` is dimensionless, `self._unit` is `None` and this throws a TypeError